python/ray/data/dataset.py

```diff
@@ -2756,11 +2733,24 @@ def fully_executed(self) -> "Dataset[T]":
         ds._set_uuid(self._get_uuid())
         return ds
 
-    @DeveloperAPI
     def stats(self) -> str:
         """Returns a string containing execution timing information."""
         return self._plan.stats().summary_string()
 
+    @DeveloperAPI
+    def get_internal_block_refs(self) -> List[ObjectRef[Block]]:
+        """Get a list of references to the underlying blocks of this dataset.
+
+        This function can be used for zero-copy access to the data. It blocks
+        until the underlying blocks are computed.
+
+        Time complexity: O(1)
+
+        Returns:
+            A list of references to this dataset's blocks.
+        """
+        return self._plan.execute().get_blocks()
+
     def _experimental_lazy(self) -> "Dataset[T]":
```

Comment:
Should we de-experimentalize lazy evaluation before GA?

Reply:
Good question. I'm leaning towards no, given that we have read task fusion enabled at least, and most of the critical use cases involve pipelines. But we can revisit based on user needs.